x-pack/plugins/cases/public/components/case_view/case_view_metrics.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import styled from 'styled-components';
+import { EuiFlexGroup, EuiFlexItem, EuiLoadingSpinner, EuiPanel } from '@elastic/eui';
+import { CaseMetrics, CaseMetricsFeature } from '../../../common/ui';
+import {
+  ASSOCIATED_HOSTS_METRIC,
+  ASSOCIATED_USERS_METRIC,
+  TOTAL_ALERTS_METRIC,
+  TOTAL_CONNECTORS_METRIC,
+} from './translations';
+
+const MetricValue = styled(EuiFlexItem)`
+  font-size: ${({ theme }) => theme.eui.euiSizeL};
+  font-weight: bold;
+`;
+
+export interface CaseViewMetricsProps {
+  metrics: CaseMetrics | null;
+  features: CaseMetricsFeature[];
+  isLoading: boolean;
+}
+
+type MetricItems = Array<{ title: string; value: number }>;
+
+const useMetricItems = (
+  metrics: CaseMetrics | null,
+  features: CaseMetricsFeature[]
+): MetricItems => {
+  const { alerts, connectors } = metrics ?? {};
+  const totalConnectors = connectors?.length ?? 0;
+  const alertsCount = alerts?.count ?? 0;
+  const totalAlertUsers = alerts?.users?.total ?? 0;
+  const totalAlertHosts = alerts?.hosts?.total ?? 0;
+
+  const metricItems = useMemo<MetricItems>(() => {
```

Comment:
nit: I think this can be simplified as:

```
const items = {
		  'alerts.count': { title: TOTAL_ALERTS_METRIC, value: alerts?.count ?? 0 },
	    'alerts.users': { title: ASSOCIATED_USERS_METRIC, value: alerts?.users?.total ?? 0 },
	    'alerts.hosts': { title: ASSOCIATED_HOSTS_METRIC, value: alerts?.hosts?.total ?? 0 },
	    'connectors': { title: TOTAL_CONNECTORS_METRIC, value: connectors?.length ?? 0 },
}

const metricItems = useMemo<MetricItems>(() => { 
	  return Object.keys(items).reduce((metricItems, key) => [
		  ...metricItems,
	    ...(features.includes(key) ? [items[key]] : [])
	  ], [])
}, [...])
```